### PR TITLE
i18n(fr) : Internationalization in French is here!

### PR DESCRIPTION
**Pull request by thomasbnt**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

Add all documentations translated in French language.

### CODE REVIEW

File: docs/src/content/docs/fr/environmental-impact.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+---
+title: Documents écologiques
+description: Découvrez comment Starlight peut vous aider à créer des sites de documentation plus écologiques et à réduire votre empreinte carbone.
```
Comment (pioupia) [inline docs/src/content/docs/fr/environmental-impact.md:3]: Is saying "des sites de documentation" is really necessary ? We can replace that by "des documentations", no ?

**Review by pioupia — APPROVED** [review]

File: docs/src/content/docs/fr/environmental-impact.md (hunk 1 above)
Comment (thomasbnt) [reply docs/src/content/docs/fr/environmental-impact.md:3]: Yes!

File: docs/src/content/docs/fr/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Cela créera un nouveau [répertoire de projet](/fr/guides/projet-structure/) avec tous les fichiers et configurations nécessaires pour votre site.
+
+:::tip[Voir en action]
+Essayez Starlight dans votre navigateur :
+[ouvrir le modèle sur StackBlitz](https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics).
+:::
+
+## Créer du contenu avec Starlight
+
+Starlight est prêt pour que vous puissiez ajouter du nouveau contenu, ou apporter vos fichiers existants !
```
Comment (iverly) [inline docs/src/content/docs/fr/getting-started.mdx:55]:
```suggestion
Starlight est conçu pour que vous puissiez ajouter du nouveau contenu, ou apporter vos fichiers existants !
```

think it's better ?

**Comment by thomasbnt** [comment]
Thanks @iverly !

**Review by iverly — APPROVED** [review]

File: packages/starlight/translations/fr.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "languageSelect.accessibleLabel": "Selectionner la langue",
+    "menuButton.accessibleLabel": "Menu",
+    "sidebarNav.accessibleLabel": "Main",
+    "tableOfContents.onThisPage": "Sur cette page",
+    "tableOfContents.overview": "Vue d'ensemble",
+    "i18n.untranslatedContent": "Ce contenu n'est pas encore disponible dans votre langue.",
+    "page.editLink": "Editer la page",
+    "page.lastUpdated": "Dernière mise à jour :",
+    "page.previousLink": "Suivant",
+    "page.nextLink": "Précédent"
```
Comment (sarah11918) [inline packages/starlight/translations/fr.json:19]:
```suggestion
    "page.previousLink": "Précédent",
    "page.nextLink": "Suivant"
```
Oui?  :smile:

**Pull request merged**